mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp

```diff
@@ -2077,77 +2083,81 @@ int main(int argc, char **argv) {
+  if (dataTypeStr == "f32") {
     dataType = builder.getF32Type();
-  } else if (tensorDataType == "f16") {
+  } else if (dataTypeStr == "f16") {
     dataType = builder.getF16Type();
-  } else if (tensorDataType == "bf16") {
+  } else if (dataTypeStr == "bf16") {
     dataType = builder.getIntegerType(16);
   }
 
-  correctParameters();
-  populateDefaults();
-
-  Conv2dGenerator conv2dGenerator;
-  // Determine dimensions.
-  SmallVector<int64_t, 5> filterDimension;
-  SmallVector<int64_t, 5> inputDimension;
-  SmallVector<int64_t, 5> outputDimension;
-  conv2dGenerator.parseConvDims(
-      inputLayout, outputLayout, filterLayout, groupSize, batchSize,
-      inputChannel, inputHeight, inputWidth, outputChannel, outputHeight,
-      outputWidth, filterWidth, filterHeight, filterDimension, inputDimension,
-      outputDimension);
-
   // Populate the module.
-  std::string kernelName;
   if (!populateCpuConvolution.getValue()) {
-    if (failed(conv2dGenerator.genConvModule(
-            arch.getValue(), num_cu.getValue(), operation.getValue(),
-            inputLayout, outputLayout, filterLayout, filterDimension,
-            inputDimension, outputDimension, dilationHeight.getValue(),
-            dilationWidth.getValue(), strideHeight.getValue(),
-            strideWidth.getValue(), paddingHeight.getValue(),
-            paddingWidth.getValue(), module, builder, kernelName, dataType,
-            xdlopsV2.getValue()))) {
+    if (failed(conv2dGenerator.genConvModule(module, builder))) {
       llvm::errs() << "Module population failed.\n";
       exit(1);
     }
   }
 
+  const auto &genConfig = conv2dGenerator.getConfig();
+
   // populate host harness and host validation.
   if (populateValidation.getValue()) {
-    if (failed(populateValidationLogic(module, builder, context,
-                                       filterDimension, inputDimension,
-                                       outputDimension, dataType))) {
+    if (failed(populateValidationLogic(
+            module, builder, context, genConfig.filterDimension,
+            genConfig.inputDimension, genConfig.outputDimension, dataType))) {
       llvm::errs() << "Host validation populated failed.\n";
       exit(1);
     }
   }
 
   // populate CPU convolution and print the results.
   if (populateCpuConvolution.getValue()) {
-    if (failed(populateCpuConvolutionLogic(module, builder, context,
-                                           filterDimension, inputDimension,
-                                           outputDimension))) {
+    if (failed(populateCpuConvolutionLogic(
+            module, builder, context, genConfig.filterDimension,
+            genConfig.inputDimension, genConfig.outputDimension))) {
       llvm::errs() << "Cpu Convolution populated failed.\n";
       exit(1);
     }
   }
 
   // Apply passes.
-  if (failed(runMLIRPasses(module, passPipeline, kernelName))) {
+  if (failed(runMLIRPasses(module, passPipeline, genConfig.kernelName))) {
```

Comment:
Now, most of the other use cases you are using only dimensions only, which is good because we know for sure the dimensions will be constructed in the `genConfig`.

Now in this instance, what if we build a partially constructed config but we ended up using its un-constructed `kernelName`?